### ISSUE TITLE
Several sensors not working with HA Core 2024.1.2

### ISSUE THREAD

**Issue by Arne1346** [pail23/stiebel_eltron_isg_component#168]
After upgrading to HA Core 2024.1.2 several sensors no longer provide data - e.g.

```
Stiebel Eltron ISG Actual Temperature HK 1
Stiebel Eltron ISG Outdoor Temperature
Stiebel Eltron ISG Flow Temperature
Stiebel Eltron ISG Actual Temperature Buffer
Stiebel Eltron ISG Actual Temperature Water
```

and more.

Others (e.g. `Stiebel Eltron ISG Is heating, Stiebel Eltron ISG Heating Curve Rise HK1, Stiebel Eltron ISG Compressor`) continued working properly.

After restoring backup and downgraded to HA Core 2023.12.4 all sensors are working again properly

**Comment by pail23** [comment]
Hi @Arne1346 

Which version of the Stiebel Eltron component do you use? Would you mind upload the diagnostic information? This would help investigating the issue.

**Comment by werpu** [comment]
I have the same issue, i checked the logs and some internal api seems to have broken the sensor read, I see following errors:

fetching Stiebel Eltron ISG data in 0.012 seconds (success: True)
2024-01-11 08:40:53.533 DEBUG (MainThread) [custom_components.stiebel_eltron_isg] Finished fetching Stiebel Eltron ISG data in 0.011 seconds (success: True)
2024-01-11 08:43:37.835 DEBUG (MainThread) [custom_components.stiebel_eltron_isg] Finished fetching Stiebel Eltron ISG data in 0.011 seconds (success: True)
2024-01-11 08:43:37.848 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up stiebel_eltron_isg platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 360, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/stiebel_eltron_isg/sensor.py", line 365, in async_setup_entry
    description.state_class = STATE_CLASS_TOTAL_INCREASING
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'state_class'
2024-01-11 08:47:56.301 DEBUG (MainThread) [custom_components.stiebel_eltron_isg] Finished fetching Stiebel Eltron ISG data in 0.032 seconds (success: True)
2024-01-11 08:47:56.336 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up stiebel_eltron_isg platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 360, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/stiebel_eltron_isg/sensor.py", line 365, in async_setup_entry
    description.state_class = STATE_CLASS_TOTAL_INCREASING
    ^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __setattr__
dataclasses.FrozenInstanceError: cannot assign to field 'state_class'

Seems like an attribute has changed!

**Comment by pail23** [comment]
In order to see the version, you need to download the diagnosis information: 
<img width="253" alt="image" src="https://github.com/pail23/stiebel_eltron_isg_component/assets/13279715/c7d9f592-c90c-48a0-80ad-7ed742a4c537">
When you open the file and search for stiebel_eltron, you will find something like this:
```
    "stiebel_eltron_isg": {
      "version": "2024.1.1",
      "requirements": [
        "pymodbus>=3.5.1"
      ]
    },
```
It would be very helpful if you could upload the file here.

**Comment by gargar69** [comment]
Below is the log error from Home Assistant. It's all the sensors. The device class used has been deprecated. 

Logger: homeassistant.const
Source: helpers/deprecation.py:205
First occurred: 13:40:45 (2 occurrences)
Last logged: 13:40:45

DEVICE_CLASS_ENERGY was used from stiebel_eltron_isg, this is a deprecated constant which will be removed in HA Core 2025.1. Use SensorDeviceClass.ENERGY instead, please create a bug report at https://github.com/pail23/stiebel_eltron_isg/issues
ENERGY_KILO_WATT_HOUR was used from stiebel_eltron_isg, this is a deprecated constant which will be removed in HA Core 2025.1. Use UnitOfEnergy.KILO_WATT_HOUR instead, please create a bug report at https://github.com/pail23/stiebel_eltron_isg/issues

**Comment by pail23** [comment]
This error has been already fixed in 2024.1.1. In case you still see this error into the latest version I would appreciate if you could post your diagnosis information here. Thanks.

**Comment by gargar69** [comment]
Thanks Paul. Nice job, very prompt. It worked first time.
GAR

**Comment by pail23** [comment]
Fixed in 2024.1.1
